app/javascript/mastodon/features/home_timeline/index.jsx

```diff
@@ -28,12 +32,43 @@ const messages = defineMessages({
   hide_announcements: { id: 'home.hide_announcements', defaultMessage: 'Hide announcements' },
 });
 
+const getHomeFeedSpeed = createSelector([
+  state => state.getIn(['timelines', 'home', 'items'], ImmutableList()).take(20).map(id => state.getIn(['statuses', id])),
+], statuses => {
+  const uniqueAccountIds = (new Set(statuses.map(status => status.get('account')).toArray())).size;
+
+  const precision = 20 * 60 * 1000; // 20 minutes
+
+  const frequency = Object.values(statuses.map(status => new Date(status.get('created_at'))).reduce((acc, cur) => {
+    const bucket = Math.floor(cur.getTime() / precision);
+
+    if (acc[bucket]) {
+      acc[bucket] += 1;
+    } else {
+      acc[bucket] = 1;
+    }
+
+    return acc;
+  }, {})).reduce((p, c, i) => p + (c - p) / (i + 1), 0);
```

Comment:
This bit is honestly hard to follow and I'm not quite sure what it's meant to compute.

I think the following does the same thing while being significantly easier to read.
```suggestion
  const tally = Object.values(statuses.map(status => new Date(status.get('created_at'))).reduce((tally, created_at) => {
    const bucket = Math.floor(created_at.getTime() / precision);
    tally[bucket] = (tally[bucket] || 0) + 1;
    return tally;
  }, {}));
  const values = Object.values(tally);
  const frequency = values.length ? values.reduce((total, value) => total + value) / values.length : 0;
```

But is that really what you *want* to compute? This is the average number of posts per bucket of 20 minutes, but not counting empty buckets.